src/expressions/tests/CombinatingTest.ts

```diff
@@ -0,0 +1,65 @@
+import Value from '../dataTypes/Value';
+import DynamicContext from '../DynamicContext';
+import ExecutionParameters from '../ExecutionParameters';
+import {} from '../Expression';
+import Specificity from '../Specificity';
+import StaticContext from '../StaticContext';
+import { Bucket, unionBucket } from '../util/Bucket';
+import TestAbstractExpression from './TestAbstractExpression';
+
+/**
+ * Combinating test, used to process `descendant::(a|b|c)` selectors
+ */
+class CombinatingTest extends TestAbstractExpression {
```

Comment:
The spec appears to call this a "UnionNodeTest" (https://qt4cg.org/specifications/xquery-40/xpath-40.html#prod-xpath40-UnionNodeTest). Let's use that same terminology here?

Reply:
Used that!